libmscore/layout.cpp

```diff
@@ -3911,8 +3911,13 @@ void LayoutContext::layout()
             prevSystem  = 0;
             }
       if (!curSystem) {
+            // The end of the score. The remaining systems are not needed...
+            qDeleteAll(systemList);
+            systemList.clear();
+            // ...and the remaining pages too
             while (score->npages() > curPage)        // Remove not needed pages. TODO: make undoable:
-                  score->pages().takeLast();
+                  // Why should that be undoable? All the pages will be rebuilt at layout...
```

Comment:
Second part of this comment is the answer to the first one, isn't it? Let's make it clear. Just use statement because we won't understand the meaning of this internal dialogue in a year :)

Reply:
I think the second part is the answer on why it shouldn't be undoable. Reusing those pages after undo operation should be very complicated given that the previous pages and systems have already changed during layout and may not return to the same state after re-layout after that undo operation. The result of layout currently does not depend solely on the state of the score. Even if we want to be able to undo re-layout this should probably be done not by preserving only part of the pages, or layout engine will need to change much before that.

Reply:
Sure, I understand. I'm asking about rephrasing it to something like "this should be undoable because blablabla".

Reply:
Updated according to [this comment](https://github.com/musescore/MuseScore/pull/4119#issuecomment-437298263).